org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/AbstractPosition.java

```diff
@@ -113,13 +113,13 @@ public final boolean equals(Object obj, boolean checkStep) {
 		if (!iours.equals(itheirs)) return false;
 
 		List<List<String>> ourDimNames = getDimensionNames().stream()
-				.map(c -> new ArrayList<String>(c))
+				.map(ArrayList::new)
 				.collect(Collectors.toList());
-		ourDimNames.forEach(c -> Collections.sort(c));
+		ourDimNames.forEach(Collections::sort);
 		List<List<String>> theirDimNames = getDimensionNames((IPosition) obj).stream()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "getDimensionNames" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)